File: FerreteriaTuerKa/Models/DetalleVenta.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace FerreteriaTuerKa.Models
{
    [Table("detalleventas")]
    public class DetalleVenta
    {
        [Key]
        public int DetalleVentaId { get; set; }

        [Required]
        public int VentaId { get; set; }

        public Venta? Venta { get; set; }

        [Required]
        public int ProductoId { get; set; }

        public Producto? Producto { get; set; }

        [Required]
        public int Cantidad { get; set; }

        [Required]
        public decimal PrecioUnitario { get; set; }

        [Required]
        public decimal Subtotal { get; set; }
    }
}

File: ferreteriaTuerKa/src/main/java/org/example/ferreteriatuerka/model/DetalleVenta.java
package org.example.ferreteriatuerka.model;

public class DetalleVenta {
    private int detalleVentaId;
    private int ventaId;
    private int productoId;
    private String nombreProducto;
    private int cantidad;
    private double precioUnitario;
    private double subtotal;

    public DetalleVenta(){}
    public DetalleVenta(int detalleVentaId, int ventaId, int productoId, String nombreProducto,
                        int cantidad, double precioUnitario, double subtotal, String emailCliente){
        this.detalleVentaId = detalleVentaId;
        this.ventaId = ventaId;
        this.productoId = productoId;
        this.nombreProducto = nombreProducto;
        this.cantidad = cantidad;
        this.precioUnitario = precioUnitario;
        this.subtotal = subtotal;
    }


    public int getProductoId(){ return productoId; }
    public void setProductoId(int productoId){ this.productoId = productoId; }

    public String getNombreProducto(){ return nombreProducto; }
    public void setNombreProducto(String nombreProducto){ this.nombreProducto = nombreProducto; }

    public int getCantidad(){ return cantidad; }
    public void setCantidad(int cantidad){ this.cantidad = cantidad; }

    public double getPrecioUnitario(){ return precioUnitario; }
    public void setPrecioUnitario(double precioUnitario){ this.precioUnitario = precioUnitario; }

    public double getSubtotal(){ return subtotal; }
    public void setSubtotal(double subtotal){ this.subtotal = subtotal; }
}